### ISSUE TITLE
Building a React Frontend

### ISSUE THREAD

**Issue by jspizziri** [garethbjohnson/gutendex#10]
Hey, first thanks for building and hosting a sane API for the Gutenberg project.

**TL;DR;**

Check out the simple react app I threw together in a few hours: https://gutenberg.netlify.com/

----

I wanted to let you know I recently embarked on a project to overhaul the Gutenberg UI, and was considering building both a frontend and API (which is when I encountered your project 🎉 ). Using your API I banged out a **SUPER** rough React frontend in a couple of hours. It needs a lot more work and it clearly has a lot of functions that are simply mocked out right now, but the popularity slider is pulling from your API and searching on a string will return the first page of results from your API (haven't done infinite scrolling or anything yet).

- I threw it up on Netlify here: https://gutenberg.netlify.com/
- You can check out the source code here: https://gitlab.com/epithet/gutenberg-org

Let me know if you'd be interested in collaborating in some way!

**Comment by garethbjohnson** [comment]
Hey, that looks pretty cool!

That's the kind of app I had in mind when I made Gutendex. I'd love to see a modern UI like that for Project Gutenberg – something like Netflix for classic books. (I actually spent a few weeks making my own site like yours, but naively focused on building something like [Epub.js](https://github.com/futurepress/epub.js) from scratch, and I lost steam. I have some leftover ideas, though.)

I am interested in helping somehow. I've done a bunch of TypeScript React before, so I can even contribute there. However, I've never collaborated online on something like that before, and I'm not much of an organizer. If you have an idea of what I could do to help, though, let me know. 🙂

**Comment by jspizziri** [comment]
@garethbjohnson ,

Thanks for the response. If you want to collaborate on the frontend I can create some tickets of what needs to be done and you can just take a crack at some and submit a PR/MR. I think there's also a good chance that there would need to be some additional functionality added to the API (if that's possible/feasible). In either case, we can just create some tickets regarding what needs to be done and then chip away at them.

Thoughts?

**Comment by jspizziri** [comment]
I created a public Trello board here:

https://trello.com/b/s4beyzYG/gutenbergorg

If you'd like to join then send me your Trello handle and I'll add you to the board.

**Comment by garethbjohnson** [comment]
That sounds great! I would be happy to add Gutendex features. For breaking changes, I can add API versioning. However, I would like to keep functionality specific to Project Gutenberg and widely applicable. If we want more, though, I can make another API that uses Gutendex.

My Trello handle is `garethjohnson16`.

**Comment by jspizziri** [comment]
Great! That all sounds good to me. Just added you to the Trello project. Please feel free to let me know what your ideas are too and we can discuss on the board itself.
